src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,255 @@
+package duke;
+import java.util.Scanner;
+
+/**
+ * The Parser class deals with making sense of the user command.
+ */
+public class Parser {
+    /** A user interface to interact with the user */
+    private Ui ui;
+    /** The command input by the user */
+    private String command;
+    private static final String LINE = "____________________________________________________________\n";
+
+    /**
+     * Instantiates the Parser object to deal with user input.
+     *
+     * @param command The command input by the user.
+     * @param ui
+     */
+    public Parser(String command, Ui ui) {
+        this.command = command;
+        this.ui = ui;
+    }
+
+    /**
+     * Reads the input given by the user and determines the commands to be carried out.
+     *
+     * @param taskList A collection of tasks.
+     * @return false if command closes the Parser, true otherwise.
+     */
+    public boolean readCommand(TaskList taskList) {
+        Scanner sc = new Scanner(command);
+        String input = sc.nextLine();
+        String[] inputArray = input.split(" ");
+        String first = inputArray[0];
+
+        try {
+            if (input.equals("bye")) {
+                return false;
+            } else if (input.equals("list")) {
+                listTasks(taskList);
+            } else if (first.equals("unmark")) {
+                unmarkTask(input, taskList);
+            } else if (first.equals("mark")) {
+                markTask(input, taskList);
+            } else if (first.equals("todo")) {
+                addTask(input, taskList);
+            } else if (first.equals("deadline")) {
+                addTask(input, taskList);
+            } else if (first.equals("event")) {
+                addTask(input, taskList);
+            } else if (first.equals("delete")) {
+                deleteTask(input, taskList);
+            } else if (first.equals("find")) {
+                findTask(input, taskList);
+            } else {
+                throw new DukeException(LINE + "☹ OOPS!!! I'm sorry, but I don't know what that means :(\n"
+                        + LINE);
+            }
+        } catch (DukeException e) {
+            System.out.println(e.getMessage());
+        } catch (NumberFormatException e) {
+            System.out.println(LINE + "☹ OOPS!!! Please make sure you include the proper index.\n" + LINE);
+        } catch (StringIndexOutOfBoundsException e) {
+            System.out.println(LINE + "☹ OOPS!!! Please make sure you include the proper index.\n" + LINE);
+        }
+        return true;
+    }
+
+    /**
+     * Returns the lists of tasks onto the console.
+     *
+     * @param taskList A collection of Task objects.
+     */
+    public static void listTasks(TaskList taskList) {
```

Comment:
Just curious if it is possible for this method (as well as the other Parser methods) to be private and non-static, as it only seems to be referenced in the readCommand method which is in Parser anyway.